lxd/resources/system.go

```diff
@@ -26,7 +30,14 @@ func GetSystem() (*api.ResourcesSystem, error) {
 	system.Type = systemType
 
 	if !sysfsExists(sysClassDMIID) {
-		return &system, nil
+		system := getSystemFromLshw()
+		if system == nil {
+			system = &api.ResourcesSystem{}
```

Comment:
On line 30 above the system.Type is set based on `systemGetType()` return value.
Previously if `!sysfsExists(sysClassDMIID)` we would have returned `system` with `Type` still set to that value:

```
		return &system, nil
```

But now with this change, you are wiping out that value and reinitalising `system` as empty.

I think it would be better to keep the old return statement of:

```
		return &system, nil
```

What do you think?

Reply:
I don't see a real issue. Do you prefer this?
```go
	if !sysfsExists(sysClassDMIID) {
		lshwSystem := getSystemFromLshw()
		if lshwSystem == nil {
			return &system, nil
		}

		lshwSystem.Type = systemType

		return lshwSystem, nil
	}
```

Reply:
Yes I think that is better because if both the sysfs and lshw command dont exist the system type from `systemGetType` will still be returned.